syntax = "proto2";

package rv.protos;

message MachineLearning {
    enum Task {
        // Only object detection is currently implemented.
        OBJECT_DETECTION = 1;
        CLASSIFICATION = 2;
        REGRESSION = 3;
        SEMANTIC_SEGMENTATION = 4;
    }

    enum Backend {
        TF_OBJECT_DETECTION_API = 1;
    }

    required Task task = 1;
    required Backend backend = 2;
}
